server/src/main/java/org/candlepin/dto/api/v1/PoolDTO.java

```diff
@@ -0,0 +1,1846 @@
+     * If the derived product attributes have not yet been defined, this method returns null.
+     * <p></p>
+     * Note that the collection returned by this method is a view of the collection backing this
+     * set of derived product attributes. Elements cannot be added to the collection,
+     * but elements may be removed.
+     * Changes made to the collection will be reflected by this pool DTO instance.
+     *
+     * @return the derived product attributes associated with this pool,
+     * or null if they have not yet been defined.
+     */
+    @JsonSerialize(using = CandlepinLegacyAttributeSerializer.class)
+    public Map<String, String> getDerivedProductAttributes() {
+        return this.derivedProductAttributes != null ?
+            new MapView<String, String>(this.derivedProductAttributes) : null;
+    }
+
+    /**
+     * Sets the derived product attributes for this pool DTO.
+     *
+     * @param derivedProductAttributes
+     *  A map of derived product attribute key, value pairs to assign to this pool DTO,
+     *  or null to clear the existing ones
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    @JsonDeserialize(using = CandlepinAttributeDeserializer.class)
+    public PoolDTO setDerivedProductAttributes(Map<String, String> derivedProductAttributes) {
+        if (derivedProductAttributes != null) {
+            if (this.derivedProductAttributes == null) {
+                this.derivedProductAttributes = new HashMap<String, String>();
+            }
+            else {
+                this.derivedProductAttributes.clear();
+            }
+            this.derivedProductAttributes.putAll(derivedProductAttributes);
+        }
+        else {
+            this.derivedProductAttributes = null;
+        }
+        return this;
+    }
+
+    /**
+     * Returns the derived product id of this pool.
+     *
+     * @return the derived product id of this pool.
+     */
+    public String getDerivedProductId() {
+        return derivedProductId;
+    }
+
+    /**
+     * Sets the derived product id of this pool.
+     *
+     * @param derivedProductId set the derived product id of this pool.
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    public PoolDTO setDerivedProductId(String derivedProductId) {
+        this.derivedProductId = derivedProductId;
+        return this;
+    }
+
+    /**
+     * Returns the derived product name of this pool.
+     *
+     * @return the derived product name of this pool.
+     */
+    @JsonProperty
+    public String getDerivedProductName() {
+        return derivedProductName;
+    }
+
+    /**
+     * Sets the derived product name of this pool.
+     *
+     * @param derivedProductName set the derived product name of this pool.
+     *
+     * @return a reference to this PoolDTO object.
+     */
+    @JsonIgnore
+    public PoolDTO setDerivedProductName(String derivedProductName) {
```

Comment:
Same with derived product. We need only the derived product object or the ID+name, not both.